src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    public static final int INITIAL_SIZE = 10;
```

Comment:
It's good to see that you've defined a constant for the initial size. However, the name `INITIAL_SIZE` could be more descriptive, such as `INITIAL_CAPACITY`, to better convey the purpose of this value.